examples/imagenet/imagenet_lib.py

```diff
@@ -123,7 +91,7 @@ def step_fn(step):
 
 def train_step(state, batch, learning_rate_fn):
   """Perform a single training step."""
-  def loss_fn(model):
+  def _loss_fn(model):
```

Comment:
I think up to now we haven't used underscores for functions in our codebase, so I would avoid it here as well for consistency.

Reply:
Or more generally, a pull request should only change things that pertain to that particular pull request.

If we think we should change our naming convention, we can discuss that but we shouldn't combine that with a pull request refactoring an example for sake of testing.

Reply:
I used this convention as this file had a similar naming convention (def _prepare(x)) in `prepare_tf_data`. Changed it back and also changed _prepare -> prepare. 

@avital yeah, I think that makes sense. Since this was a huge refactoring, I just put them the smaller ones in as well, probably wasn't a good idea :)